### PR TITLE
feat(sdk-coin-avaxp): Verify import in P

### PR DESCRIPTION
**Pull request by colin-y-wu**
Ticket: BG-55595

### CODE REVIEW

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 1]
```diff
@@ -121,6 +122,22 @@ export class AvaxP extends BaseCoin {
     }
   }
 
+  validateImportInP(explainedTxInputs: AvaxpLib.AvaxpEntry[], txParams: AvaxpTransactionParams): void {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/src/avaxp.ts:125]: Can we get documentation for this method?

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

**Review by CapnMigraine — COMMENTED** [review]
Please fix the commit message and PR description to reference ticket [BG-55595](https://bitgoinc.atlassian.net/browse/BG-55595)

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 2]
```diff
@@ -163,14 +180,19 @@ export class AvaxP extends BaseCoin {
         break;
       case TransactionType.Import:
         if (tx.isTransactionForCChain) {
+          // ImportInC
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:183]: clarification-nit:
```suggestion
          // Import to C-chain
```

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 3]
```diff
@@ -163,14 +180,19 @@ export class AvaxP extends BaseCoin {
...
           if (
             (params.txParams.recipients && params.txParams.recipients.length !== 0) ||
             explainedTx.outputs.length !== 1
           ) {
             throw new Error('Expected 1 output in import txn and does not require recipients');
           }
         } else {
-          // TODO: BG-55595 - verify import txn on P chain
+          // ImportInP
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:191]: clarification-nit:
```suggestion
          // Import to P-chain
```

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 4]
```diff
@@ -163,14 +180,19 @@ export class AvaxP extends BaseCoin {
...
+          if (explainedTx.outputs.length !== 1) {
+            throw new Error('Expected 1 output in import txn');
+          }
+          this.validateImportInP(explainedTx.inputs, params.txParams);
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:195]: consistency-nit:
Please avoid using export "in" and import "in"
```suggestion
          this.validateImportTx(explainedTx.inputs, params.txParams);
```

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 5]
```diff
@@ -473,6 +473,44 @@ describe('Avaxp', function () {
       isTransactionVerified.should.equal(true);
     });
 
+    it('should succeed to verify import in P transaction', async () => {
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:476]:
```suggestion
    it('should succeed to verify import to P transaction', async () => {
```

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 6]
```diff
@@ -473,6 +473,44 @@ describe('Avaxp', function () {
...
+        type: 'Import',
+        locktime: 0,
+        memo: undefined,
+        unspents: ['e8ixKnba52yufXrTVKrTXVQTj5cd5e6o6Lc3rVkhahDGEs72L:0'],
+      };
+      const isTransactionVerified = await basecoin.verifyTransaction({ txParams, txPrebuild });
+      isTransactionVerified.should.equal(true);
+    });
+
+    it('should fail to verify import in P transaction with wrong unspents', async () => {
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:492]:
```suggestion
    it('should fail to verify import to P transaction with wrong unspents', async () => {
```

File: modules/sdk-coin-avaxp/src/avaxp.ts (hunk 1 above)
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:125]: consistency-nit:
```suggestion
  validateImportTx(explainedTxInputs: AvaxpLib.AvaxpEntry[], txParams: AvaxpTransactionParams): void {
```

**Review by CapnMigraine — CHANGES_REQUESTED** [review]

**Review by CapnMigraine — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**